### ISSUE TITLE
Regularisation losses in nested layers

### ISSUE THREAD

**Issue by georgesterpu** [tensorflow/tensorflow#33147]
**System information**
- Have I written custom code: Yes
- OS Platform and Distribution: Manjaro testing, x86_64
- TensorFlow installed from (source or binary): pypi binary
- TensorFlow version (use command below): v2.0.0-rc2-26-g64c3d38 2.0.0
- Python version: 3.7.4

**Describe the current behavior**
TensorFlow returns duplicated regularisation losses for layers which hold references to regularised variables built in other layers.

**Describe the expected behavior**
Return a single regularisation loss per variable.

**Code to reproduce the issue**
```
import tensorflow as tf


class A(tf.keras.layers.Layer):
    def __init__(self, layer):
        super(A, self).__init__()
        self.layer = layer

    def call(self, inputs):
        return self.layer(inputs)


class B(tf.keras.layers.Layer):
    def __init__(self):
        super(B, self).__init__()
        self.obj = tf.keras.layers.Dense(13, kernel_regularizer=tf.keras.regularizers.l1(5))
        self.layerB = A(self.obj)

    def call(self, inputs):
        return self.layerB(inputs)

model = B()

output = model(tf.ones([5, 10]))
print(len(model.losses))
```
Once we rename self.obj to obj, i.e. not saving it as a class member, we end up with a single regularisation loss.

**Other info / logs**
More info and logs in https://github.com/tensorflow/addons/issues/577
Plus @guillaumekln 's response https://github.com/tensorflow/addons/issues/577#issuecomment-539530375

**Comment by oanush** [comment]
Issue replicating with TF-2.0, kindly find the [gist](https://colab.sandbox.google.com/gist/oanush/2b96e1dc23de60e6992be9f598a74b53/33147.ipynb) of colab.Thanks!

**Comment by georgesterpu** [comment]
https://github.com/tensorflow/tensorflow/commit/e2f31134bd8aaf7cdc8b6f157f4fee3f1e069e0c

**Comment by saikumarchalla** [comment]
@georgesterpu Issue seems to be fixed in latest TF version '2.2.0-rc3'. Could you please confirm and close the issue if it is resolved for you. Thanks!
